examples/proxy/lib.rs

```diff
@@ -9,53 +9,93 @@
 //!     `forward_to` address at any point.
 //!
 //! Using this pattern it is possible to implement upgradable contracts.
-//!
-//! Note though that the contract to which calls are forwarded still
-//! contains it's own state.
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
 use ink_lang as ink;
 
 #[ink::contract]
 pub mod proxy {
-    /// A simple proxy contract.
-    #[ink(storage)]
-    pub struct Proxy {
-        /// The `AccountId` of a contract where any call that does not match a
+    use ink_env::call::DelegateCall;
+    use ink_primitives::{
+        Key,
+        KeyPtr,
+    };
+    use ink_storage::traits::SpreadLayout;
+
+    #[derive(Debug)]
+    #[cfg_attr(feature = "std", derive(ink_storage::traits::StorageLayout))]
+    struct ProxyFields {
```

Comment:
Could you add a comment with the description "Why did you move all fields to separate structure"?